### PR TITLE
Issue #362: `GitRepositorySandbox.new_repo` has a side effect of changing the current directory

### LINKED ISSUE #362: `GitRepositorySandbox.new_repo` has a side effect of changing the current directory
```python
    def new_repo(self, *args: str) -> "GitRepositorySandbox":
        os.chdir(args[0])
        opts = args[1:]
        self.execute(f"git init {' '.join(opts)}")
        return self
```

Some `os.chdir` back to the original position would be useful here... or, if the current behavior makes more sense, maybe a rename to this method?

### PR DESCRIPTION
**Pull request by amalota**

### CODE REVIEW

File: tests/mockers.py [hunk 1]
```diff
@@ -55,10 +55,13 @@ def execute(self, command: str) -> "GitRepositorySandbox":
         subprocess.check_output(command, stderr=subprocess.STDOUT, shell=True)
         return self
 
-    def new_repo(self, *args: str) -> "GitRepositorySandbox":
+    def new_repo(self, *args: str, switch_dir_to_new_repo: bool = True) -> "GitRepositorySandbox":
```
Comment (PawelLipski) [inline tests/mockers.py:58]: How many times is this method called across the project? TBH I'd prefer if `switch_dir_to_new_repo` param did NOT have a default value... let's make the behavior as explicit as possible

Comment (PawelLipski) [inline tests/mockers.py:58]: How many times is this method called across the project? TBH I'd prefer if `switch_dir_to_new_repo` param did NOT have a default value... let's make the behavior as explicit as possible

Comment (amalota) [reply tests/mockers.py:58]: It's called 45 times across tests, so a lot, because of that i would stick with default value, wdyt?

Comment (PawelLipski) [reply tests/mockers.py:58]: Huh okay... let's live with that then

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**